### ISSUE TITLE
Storage does not handle all null cases and causes app to not launch

### ISSUE THREAD

**Issue by rayyan35p** [rayyan35p/pe#6]
## Description

When there is null put into the tags section of the json file, the app does not launch.

## Expected

App should launch but empty, to indicate that there is invalid data.

## Actual

App does not even launch.

## Steps to Reproduce
1. Open `data/addressbook.json` and edit the tags to be as such:

![image.png](https://raw.githubusercontent.com/rayyan35p/pe/main/files/1875cd68-5b02-4813-a157-97968f82340d.png)

2. Launch JABPro using the terminal and notice that the app is not able to launch.

**Comment by nus-pe-script** [comment]
# Team's Response
Rejected based on the basis of **extreme user behavior**.

Firstly, the editing of the file welcome advanced users in order to manipulate the file directly. 
This would mean that a huge portion of users wont be manipulating the data file directly.  

Secondly, due to the CLI aspect of the app, there is no way for a user to give a null input. This is because all inputs are string.  
Thereby, the only introduction of null input into tag can only be done either 1. by direct access to the json data file or 2. by the developer. 
In this case, 2 is not possible since we used defensive programming in order to verify non-null tags.  
The point is, there will never be a null tag that shows up in your json file if you use the CLI. So the Hiring Manager will never be able to take reference from the data and try out a null value.  

Thirdly, the probability of a Hiring Manager using the null keyword is extremely low. Since most of the tags have  tagName and tagCategory and are formatted as Strings using `" "`, 
There is a reasonable expectation that Hiring Manager will follow such a format when keying in tags.

![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2324S1/pe/master/files/03b7f9b6-ba8d-4d1f-8d8c-b4e10bc57574.png)



Thus the assumption that a null appears in tag will be 
1. The hiring manager understands the null keyword and how to use it
2. The hiring manager is intentionally trying to test the breaking point of the application 

This deviates far too greatly from expected user behavior and would instead be a behavior of a user trying to break the application. With an extremely low probability of such an input and the logical case of using null close to 0, we therefore conclude this is a extreme user behaviour



![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2324S1/pe/master/files/022df708-8e64-4111-9033-665ac68a844c.png)
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.Rejected`]

- [x] I disagree

**Reason for disagreement:** I would like to argue that entering null is not an extreme user behavior.

There is a much more significant chance of the user entering `null` as opposed to an extreme input like a _30-digit telephone number_ which was used as an example.

Firstly, a **4**-letter word is much more easily input than an extremely long input such as the aforementioned **30**-digit number.

Secondly, there are words that can be entered(although not very nice) that are very possible, although not probable. Inputs such as `dull` may be entered as a `tag` to describe an applicant that did not appear very lively during the interview. When entering this directly in the json, there is a non-zero chance of the `n` key being pressed and spelling `null` instead.

Lastly, regarding the point of the values having to be formatted as Strings through the `"` character, there is still always a possibility that a user, advanced or not, can forget to input it, especially when the json files are being edited in text editors which may not do syntax checks.

In summary, `null` cannot be categorised as an extreme behavior due to the low level of effort and difficulty in inputting it, thus the issue should not be rejected as it is still possible to occur.


<catcher-end-of-segment><hr>
